unitpackage/collection.py

```diff
@@ -290,12 +298,222 @@ def __getitem__(self, identifier):
             ...
             KeyError: "No collection entry with identifier 'invalid_key'."
 
+        An entry from an integer::
+
+            >>> collection[0]
+            Entry('alves_2011_electrochemistry_6010_f1a_solid')
+
+            >>> collection[-1]
+            Traceback (most recent call last):
+            ...
+            IndexError: Index -1 out of range for collection with 3 entries.
+
+        A collection with a single entry from a single identifier or index::
+
+            >>> collection[['alves_2011_electrochemistry_6010_f1a_solid']]
+            [Entry('alves_2011_electrochemistry_6010_f1a_solid')]
+
+            >>> collection[[0]]
+            [Entry('alves_2011_electrochemistry_6010_f1a_solid')]
+
+        A new collection with entries selected by a slice::
+
+            >>> collection[1:2]
+            [Entry('engstfeld_2018_polycrystalline_17743_f4b_1')]
+
+            >>> collection[-1:2]
+            Traceback (most recent call last):
+            ...
+            IndexError: slice(-1, 2, None) out of range for collection.
+
+        A new collection with entries selected by a list of identifiers::
+
+            >>> collection['alves_2011_electrochemistry_6010_f1a_solid', 'engstfeld_2018_polycrystalline_17743_f4b_1']
+            [Entry('alves_2011_electrochemistry_6010_f1a_solid'), Entry('engstfeld_2018_polycrystalline_17743_f4b_1')]
+
+            >>> collection['alves_2011_electrochemistry_6010_f1a_solid', 'invalid_key']
+            [Entry('alves_2011_electrochemistry_6010_f1a_solid')]
+
+            >>> collection[['alves_2011_electrochemistry_6010_f1a_solid', 'invalid_key']]
+            [Entry('alves_2011_electrochemistry_6010_f1a_solid')]
+
+            >>> collection['invalid_key', 'invalid_key2']
+            Traceback (most recent call last):
+            ...
+            KeyError: "None of the provided ('invalid_key', 'invalid_key2'), are valid identifiers for this collection."
```

Comment:
Why only raise an exception when only invalid keys are provided and not whenever at least one invalid key is provided?